tests/test_mlx5_pp.py

```diff
@@ -0,0 +1,57 @@
+# SPDX-License-Identifier: (GPL-2.0 OR Linux-OpenIB)
+# Copyright (c) 2019 Mellanox Technologies, Inc. All rights reserved. See COPYING file
+
+"""
+Test module for mlx5 packet pacing entry allocation.
+"""
+
+from pyverbs.providers.mlx5.mlx5dv import Mlx5PP, Mlx5Context, Mlx5DVContextAttr
+from pyverbs.pyverbs_error import PyverbsRDMAError, PyverbsUserError
+import pyverbs.providers.mlx5.mlx5_enums as e
+from tests.base import RDMATestCase
+import unittest
+import errno
+
+
+class Mlx5PPRes:
+    def __init__(self, dev_name):
+        try:
+            mlx5dv_attr = Mlx5DVContextAttr(e.MLX5DV_CONTEXT_FLAGS_DEVX)
+            self.ctx = Mlx5Context(mlx5dv_attr, dev_name)
+        except PyverbsUserError as ex:
+            raise unittest.SkipTest('Could not open mlx5 context ({})'
+                                    .format(str(ex)))
+        except PyverbsRDMAError as ex:
+            if ex.error_code == errno.EOPNOTSUPP:
+                raise unittest.SkipTest('Opening mlx5 DevX context is not supported')
+            raise ex
+        self.pps = []
+
+
+class Mlx5PPTestCase(RDMATestCase):
+    def setUp(self):
+        super().setUp()
+        self.pp_res = Mlx5PPRes(self.dev_name)
+
+    def test_pp_alloc(self):
+        """
+        Allocate two packet pacing entries with the same configuration. One of
+        the entries is allocated with a dedicated index.
+        Then verify that the indexes are different and free the entries.
+        """
+        # An arbitrary valid rate limit value (in kbps)
+        rate_limit = (100).to_bytes(length=4, byteorder='big', signed=True)
```

Comment:
Please use the struct module for this

Reply:
I'll use "rate_limit = struct.pack('>I', 100)" instead.
PR will be updated.

Reply:
PR was updated as asked including kernel-headers.